### ISSUE TITLE
Pytest with xvfb plugin freeze

### ISSUE THREAD

**Issue by emcek** [The-Compiler/pytest-xvfb#31]
## Bug description

Tests are started but never end

## To Reproduce

Steps to reproduce the behavior. Something like:
1. Setup some simple qt app with one test
2. add to .gitlab-cy.yml:
```yaml
image: python:3.10
variables:
  DISPLAY: ':99.0'
  PYTEST_QT_API: 'pyqt5'

stages:
  - qt

'qt test':
  stage: qt
  script:
    - apt-get update && apt-get autoclean
    - apt-get install -y '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev ffmpeg libsm6 libxext6 xvfb
    - env
    - pip install PyQt5==5.15.7 pytest pytest-qt pytest-xvfb
    - pytest -vvvv -s tests/test_qtgui.py
```

Example of pipeline: https://gitlab.com/modding-openmw/modhelpertool/-/jobs/3019205865
In order to `pytest-xvfb` start I had to install:
```shell
apt-get install -y '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev ffmpeg libsm6 libxext6 xvfb
```

## System information
Keep example of pipeline  for the record:
```shell
$ apt-get install -y '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev ffmpeg libsm6 libxext6 xvfb
$ env
CI_PROJECT_NAMESPACE=modding-openmw
GITLAB_USER_ID=504651
CI_RUNNER_VERSION=15.4.0~beta.5.gdefc7017
FF_SKIP_NOOP_BUILD_STAGES=true
CI_SERVER_NAME=GitLab
CI_RUNNER_DESCRIPTION=2-green.shared.runners-manager.gitlab.com/default
GITLAB_USER_EMAIL=mplichta@gmail.com
CI_SERVER_REVISION=b9548e306fe
FF_USE_WINDOWS_LEGACY_PROCESS_STRATEGY=true
CI_RUNNER_EXECUTABLE_ARCH=linux/amd64
CI_REGISTRY_USER=gitlab-ci-token
CI_API_V4_URL=https://gitlab.com/api/v4
CI_REGISTRY_PASSWORD=[MASKED]
CI_RUNNER_SHORT_TOKEN=ns46NMmJ
CI_JOB_NAME=qt test
HOSTNAME=runner-ns46nmmj-project-37865482-concurrent-0
PYTHON_VERSION=3.10.7
GITLAB_USER_LOGIN=emcek
CI_PROJECT_NAME=modhelpertool
CI_PIPELINE_SOURCE=push
CI_BUILD_REF_NAME=qt_test
CI_JOB_STATUS=running
CI_PIPELINE_ID=638512982
CI_BUILD_REF_SLUG=qt-test
CI_COMMIT_REF_SLUG=qt-test
CI_SERVER=yes
CI_COMMIT_SHORT_SHA=e5da6592
FF_CMD_DISABLE_DELAYED_ERROR_LEVEL_EXPANSION=false
CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX=gitlab.com:443/modding-openmw/dependency_proxy/containers
PWD=/builds/modding-openmw/modhelpertool
CI_RUNNER_TAGS=["gce", "east-c", "linux", "ruby", "mysql", "postgres", "mongo", "git-annex", "shared", "docker"]
CI_PROJECT_PATH=modding-openmw/modhelpertool
FF_USE_NEW_BASH_EVAL_STRATEGY=false
CI_SERVER_TLS_CA_FILE=/builds/modding-openmw/modhelpertool.tmp/CI_SERVER_TLS_CA_FILE
CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX=gitlab.com:443/modding-openmw/dependency_proxy/containers
PYTHON_SETUPTOOLS_VERSION=63.2.0
CI_COMMIT_REF_PROTECTED=false
FF_USE_POWERSHELL_PATH_RESOLVER=false
CI_SERVER_VERSION_MINOR=4
CI_COMMIT_SHA=e5da659279128204fdd0c66ede1c06b1d0c66c95
HOME=/root
FF_NETWORK_PER_BUILD=false
LANG=C.UTF-8
CI_DEPENDENCY_PROXY_PASSWORD=[MASKED]
CI_PROJECT_VISIBILITY=public
CI_CONCURRENT_PROJECT_ID=0
FF_SCRIPT_SECTIONS=false
CI_COMMIT_MESSAGE=test qt
DOCKER_TLS_CERTDIR=
CI_JOB_JWT_V1=[MASKED]
CI_JOB_JWT_V2=[MASKED]
CI_BUILD_NAME=qt test
FF_USE_DIRECT_DOWNLOAD=true
CI_PAGES_DOMAIN=gitlab.io
CI_SERVER_VERSION=15.4.0-pre
GPG_KEY=A035C8C19219BA821ECEA86B64E628F8D684696D
CI_REGISTRY=registry.gitlab.com
CI_SERVER_PORT=443
FF_USE_IMPROVED_URL_MASKING=false
CI_PAGES_URL=https://modding-openmw.gitlab.io/modhelpertool
CI_PIPELINE_IID=395
CI_REPOSITORY_URL=[https://gitlab-ci-token:[MASKED]@gitlab.com/modding-openmw/modhelpertool.git](https://gitlab-ci-token:%5BMASKED%5D@gitlab.com/modding-openmw/modhelpertool.git)
CI_SERVER_URL=https://gitlab.com/
FF_ENABLE_BASH_EXIT_CODE_CHECK=false
GITLAB_FEATURES=elastic_search,ldap_group_sync,multiple_ldap_servers,seat_link,usage_quotas,repository_size_limit,admin_audit_log,auditor_user,custom_file_templates,custom_project_templates,db_load_balancing,default_branch_protection_restriction_in_groups,extended_audit_events,external_authorization_service_api_management,geo,ldap_group_sync_filter,object_storage,pages_size_limit,password_complexity,project_aliases,enterprise_templates,git_abuse_rate_limit,required_ci_templates,runner_maintenance_note,runner_upgrade_management
CI_COMMIT_DESCRIPTION=
CI_TEMPLATE_REGISTRY_HOST=registry.gitlab.com
CI_JOB_STAGE=qt
CI_PIPELINE_URL=https://gitlab.com/modding-openmw/modhelpertool/-/pipelines/638512982
CI_DEFAULT_BRANCH=main
CI_BUILD_REF=e5da659279128204fdd0c66ede1c06b1d0c66c95
CI_SERVER_VERSION_PATCH=0
CI_COMMIT_TITLE=test qt
CI_PROJECT_ROOT_NAMESPACE=modding-openmw
FF_ENABLE_JOB_CLEANUP=false
GITLAB_USER_NAME=Michal Plichta
CI_PROJECT_DIR=/builds/modding-openmw/modhelpertool
DISPLAY=:99.0
SHLVL=1
CI_RUNNER_ID=12270848
CI_PIPELINE_CREATED_AT=2022-09-13T08:39:08Z
CI_COMMIT_TIMESTAMP=2022-09-13T10:38:51+02:00
CI_DISPOSABLE_ENVIRONMENT=true
CI_JOB_JWT=[MASKED]
CI_REGISTRY_IMAGE=registry.gitlab.com/modding-openmw/modhelpertool
CI_BUILD_ID=3019205865
CI_SERVER_PROTOCOL=https
PYTHON_PIP_VERSION=22.2.2
CI_COMMIT_AUTHOR=Michał Plichta <mplichta@gmail.com>
FF_POSIXLY_CORRECT_ESCAPES=false
CI_COMMIT_REF_NAME=qt_test
CI_SERVER_HOST=gitlab.com
CI_JOB_URL=https://gitlab.com/modding-openmw/modhelpertool/-/jobs/3019205865
CI_JOB_TOKEN=[MASKED]
CI_JOB_STARTED_AT=2022-09-13T08:39:09Z
CI_CONCURRENT_ID=190
CI_BUILD_TOKEN=[MASKED]
CI_PROJECT_DESCRIPTION=A tool that helps with common tasks.
CI_COMMIT_BRANCH=qt_test
CI_PROJECT_CLASSIFICATION_LABEL=
FF_USE_LEGACY_KUBERNETES_EXECUTION_STRATEGY=false
PYTEST_QT_API=pyqt5
CI_RUNNER_REVISION=defc7017
PYTHON_GET_PIP_SHA256=5aefe6ade911d997af080b315ebcb7f882212d070465df544e1175ac2be519b4
CI_BUILD_BEFORE_SHA=c7ecc326ea015037fbb028f302d49e0f95ed7ca5
FF_KUBERNETES_HONOR_ENTRYPOINT=false
FF_USE_NEW_SHELL_ESCAPE=false
CI_DEPENDENCY_PROXY_USER=gitlab-ci-token
FF_USE_DYNAMIC_TRACE_FORCE_SEND_INTERVAL=false
FF_DISABLE_UMASK_FOR_DOCKER_EXECUTOR=false
CI_PROJECT_PATH_SLUG=modding-openmw-modhelpertool
CI_NODE_TOTAL=1
CI_BUILDS_DIR=/builds
CI_JOB_ID=3019205865
CI_PROJECT_REPOSITORY_LANGUAGES=perl,python,qmake
PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/5eaac1050023df1f5c98b173b248c260023f2278/public/get-pip.py
PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
CI_PROJECT_ID=37865482
CI=true
GITLAB_CI=true
CI_JOB_IMAGE=python:3.10
CI_COMMIT_BEFORE_SHA=c7ecc326ea015037fbb028f302d49e0f95ed7ca5
CI_PROJECT_TITLE=ModHelperTool
CI_SERVER_VERSION_MAJOR=15
CI_CONFIG_PATH=.gitlab-ci.yml
FF_USE_FASTZIP=false
CI_BUILD_STAGE=qt
CI_DEPENDENCY_PROXY_SERVER=gitlab.com:443
DOCKER_DRIVER=overlay2
CI_PROJECT_URL=https://gitlab.com/modding-openmw/modhelpertool
OLDPWD=/
_=/usr/bin/env
$ pip install -r requirements_test.txt
Installing collected packages: types-PyYAML, snowballstemmer, pyvirtualdisplay, PyQt5-Qt5, mypy-extensions, iniconfig, distlib, typing-extensions, tomli, six, PyYAML, PyQt5-sip, pyparsing, pyflakes, pydocstyle, pycodestyle, py, pluggy, platformdirs, mccabe, lxml, filelock, attrs, virtualenv, PyQt5, packaging, mypy, flake8, tox, qtpy, pytest, QtAwesome, pytest-xvfb, pytest-random, pytest-qt, pytest-mccabe
Successfully installed PyQt5-5.15.7 PyQt5-Qt5-5.15.2 PyQt5-sip-12.11.0 PyYAML-6.0 QtAwesome-1.1.1 attrs-22.1.0 distlib-0.3.6 filelock-3.8.0 flake8-5.0.4 iniconfig-1.1.1 lxml-4.9.1 mccabe-0.7.0 mypy-0.971 mypy-extensions-0.4.3 packaging-21.3 platformdirs-2.5.2 pluggy-1.0.0 py-1.11.0 pycodestyle-2.9.1 pydocstyle-6.1.1 pyflakes-2.5.0 pyparsing-3.0.9 pytest-7.1.3 pytest-mccabe-2.0 pytest-qt-4.1.0 pytest-random-0.2 pytest-xvfb-2.0.0 pyvirtualdisplay-3.0 qtpy-2.2.0 six-1.16.0 snowballstemmer-2.2.0 tomli-2.0.1 tox-3.26.0 types-PyYAML-6.0.11 typing-extensions-4.3.0 virtualenv-20.16.5
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
$ pytest -vvvv -s tests/test_qtgui.py
============================= test session starts ==============================
platform linux -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0 -- /usr/local/bin/python
cachedir: .pytest_cache
PyQt5 5.15.7 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /builds/modding-openmw/modhelpertool, configfile: setup.cfg
plugins: xvfb-2.0.0, qt-4.1.0, random-0.2, mccabe-2.0
collecting ... collected 1 item
tests/test_qtgui.py::test_switch_tes3cmd_version QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
ERROR: Job failed: execution took longer than 10m0s seconds
```

**Comment by jacobszpz** [comment]
they aint never replying bro :)

**Comment by The-Compiler** [comment]
@emcek If you're still interested in tracking this down, can you please run pytest with `-o faulthandler_timeout=10` (or whatever value seems sensible), so you get a stracktrace? See the relevant [docs](https://docs.pytest.org/en/7.1.x/how-to/failures.html#faulthandler) for details. Note however it seems very unlikely to me that this is actually an issue in the plugin.

@jacobszpz Imagine me missing one of dozens notification emails, oh the horror :scream: Things like this are what causes open source maintainers to burn out, great job.

**Comment by emcek** [comment]
@The-Compiler You are correct, probably they fix something in gitlab-runner old `15.4.0~beta.5.gdefc7017 (defc7017)` was not working but newer `16.1.0~beta.5.gf131a6a2 (f131a6a2)` is fine: 
https://gitlab.com/modding-openmw/modhelpertool/-/jobs/3019205865

**Comment by jacobszpz** [comment]
@The-Compiler my bad, thought you weren't maintaining the project any more. Thanks for the prompt replies though.
